pkg/app/piped/executor/ecs/rollback.go

```diff
@@ -127,12 +127,46 @@ func rollback(ctx context.Context, in *executor.Input, platformProviderName stri
 	}
 
 	// On rolling back, the scale of desired tasks will be set to 100 (same as the original state).
-	taskSet, err := client.CreateTaskSet(ctx, *service, *td, targetGroup, 100)
+	taskSet, err := client.CreateTaskSet(ctx, *service, *td, primaryTargetGroup, 100)
 	if err != nil {
 		in.LogPersister.Errorf("Failed to create ECS task set %s: %v", *serviceDefinition.ServiceName, err)
 		return false
 	}
 
+	// Reset routing
+	routingTrafficCfg := provider.RoutingTrafficConfig{
+		{
+			TargetGroupArn: *primaryTargetGroup.TargetGroupArn,
+			Weight:         100,
+		},
+		{
+			TargetGroupArn: *canaryTargetGroup.TargetGroupArn,
+			Weight:         0,
+		},
+	}
+
+	currListenerArns, err := client.GetListenerArns(ctx, *primaryTargetGroup)
+	if err != nil {
+		in.LogPersister.Errorf("Failed to get current active listener: %v", err)
+		return false
+	}
+
+	if err := client.ModifyListeners(ctx, currListenerArns, routingTrafficCfg); err != nil {
+		in.LogPersister.Errorf("Failed to routing traffic to CANARY variant: %v", err)
+		return false
+	}
+
+	// Delete Canary taskSet
+	canaryTaskSet, ok := in.MetadataStore.Shared().Get(canaryTaskSetARNKeyName)
+	if !ok {
+		in.LogPersister.Errorf("Unable to restore CANARY task set to clean: Not found")
+		return false
+	}
+	if err = client.DeleteTaskSet(ctx, *service, canaryTaskSet); err != nil {
+		in.LogPersister.Errorf("Failed to remove unused previous PRIMARY taskSet %s: %v", canaryTaskSet, err)
+		return false
+	}
+
```

Comment:
I think it's safer to do this after L182 (after we finish updating the rollbacked taskset as the primary taskset of the current service and remove the previous primary taskset), since it's more sense to only routing the traffic after primary tasksets are well-prepared and ready to handle incoming requests.

Reply:
- You're right, it's a secure choice to proceed 🙏

Reply:
@khanhtc1202 I fixed it! [reuse clean function](https://github.com/pipe-cd/pipecd/pull/4465/commits/40c0f29c1893e20dd7e70308cb7e52d4ed5dd46a), please review it 🙏